src/tests/nativeaot/ExportsFromReferencedAssemblies/NativeHost.cpp

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifdef TARGET_WINDOWS
+#include "windows.h"
+#else
+#include "dlfcn.h"
+#endif
+#include "stdio.h"
+#include "string.h"
+
+#ifndef TARGET_WINDOWS
+#define __stdcall
+#endif
+
+// typedef for shared lib exported methods
+typedef void(__stdcall *fptr)(int);
+
+#ifdef TARGET_WINDOWS
```

Comment:
The custom main should not be needed once you address https://github.com/dotnet/runtime/pull/83396/files#r1135986350

Reply:
Test case updated.